packages/slice-machine/src/features/customTypes/customTypesBuilder/GenerateSliceWithAiModal/SliceCard.tsx

```diff
@@ -1,56 +1,70 @@
+import { Button } from "@prismicio/editor-ui";
+
 import { Card, CardFooter, CardMedia } from "@/components/Card";
 
 interface SliceCardProps {
   slice: Slice;
+  onRetry?: () => void;
```

Comment:
on retry should be a property in the uploading error case, i.e. there's nothing to retry if the status is uploading, for instance

Reply:
I'm having a hard time getting this typed correctly. Any suggestions?

Reply:
see my suggestion before

Reply:
Thanks